execucao/next-js/v<=12/paginas-e-padroes.md

```diff
@@ -0,0 +1,233 @@
+
+Essas páginas são renderizadas na build do app, e _assets_ estáticos são gerados, incluindo seu HTML e JS. Quando o usuário acessar a página, o HTML e JS serão enviados como estão, e o usuário não precisará esperar o servidor renderizar a página ou fazer qualquer processamento.
+
+## ISR
+
+Para transformar uma página de SG para ISR, existem algumas maneiras. Uma delas é usar a função `getStaticProps` e passar um parâmetro `revalidate` para ela. Esse parâmetro é um número que representa o tempo em segundos que o Next deve esperar para revalidar a página. Por exemplo, se o parâmetro for 10, a página será revalidada a cada 10 segundos.
+
+### Exemplo com revalidação a cada 10 segundos:
+
+```jsx
+// src/pages/wISR/index.js.js
+
+// export default Pagina () {...}
+
+export async function getStaticProps() {
+  const response = await fetch("https://jsonplaceholder.typicode.com/todos/1");
+  const todos = await response.json();
+
+  return {
+    props: {
+      initialTodos: todos,
+    },
+    revalidate: 10, // Revalida a página a cada 10 segundos
+  };
+}
+```
+
+Página acima acessada pela url `http://localhost:3000/wISR`.
+
+Outra maneira de transformar uma página de SG para ISR é usando a função `getStaticPaths`. Essa função é executada no momento de build e deve retornar um objeto com um parâmetro `paths`, um array de objetos com `params`. Cada objeto dentro de `paths` representa uma página/url que será gerada no momento de build.
+
+### Exemplo de getStaticPaths:
+
+```jsx
+// src/pages/wISR/[id].js
+
+// export default Pagina () {...}
+
+export async function getStaticProps(context) {
+  // context é um objeto que sempre é passado para essas funções,
+  // contendo informações sobre a requisição
+
+  // pegando id da rota do next
+  const { id } = context.params;
+
+  const response = await fetch(
+    `https://jsonplaceholder.typicode.com/todos/${id}`
+  );
+
+  const todos = await response.json();
+
+  return {
+    props: {
+      initialTodos: todos,
+    },
+  };
+}
+
+export async function getStaticPaths() {
+  return {
+    paths: [
+      { params: { id: "1" } },
+      { params: { id: "2" } },
+      { params: { id: "3" } },
+    ],
+    fallback: "blocking",
+  };
+}
+```
+
+O parâmetro `params` é um objeto com os parâmetros que serão passados para a página. De acordo com o exemplo exemplo, se a página for acessada pela url `http://localhost:3000/wISR/1`, o objeto `params` deve ser `{ id: 1 }`. O Next irá gerar uma página para cada objeto dentro de `paths`.
+
+Além disso, o fallback determina o comportamento que o Next deve ter quando uma página não estiver gerada ainda.
+
+- Se o fallback for `false`, o Next irá retornar um erro 404.
+- Se o fallback for `true`, não lembro o que acontece. Pesquisa aí e adiciona nessa documentação, caro leitor.
```

Comment:
Pelo que eu entendi lendo [aqui](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths), se for true basicamente será renderizado uma versão de fallback da página na primeira vez que ela for carregada, e depois vai inseri-la na lista de páginas pré renderizadas